wordpress-mobile/WordPress-Android
# Issue/14128 reader related posts

Fixes #14128

This is a master PR for bringing back related posts feature with updated design. All sub-tasks can be found [here](https://github.com/wordpress-mobile/WordPress-Android/issues/14128).

To test:
All (sub-task) PRs are already reviewed. Smoke test the feature.

Test.1

1. Open a wp.com post on Reader tab.
2. Scroll to bottom of the post details page.
3. Notice that local and related posts are displayed with updated design.
4. Click a local or related post. 
5. Notice that selected related post details are shown on "Related Post" screen with a cross button.
6. Open another related post from the previous "Related Post" screen.
7. Notice that selected related post details are shown on "Related Post" screen with a cross button. 
8. Go back (either by the back navigational button or gesture on device).
9. Notice that previous related post is displayed.
10. Click cross button on "Related Post" screen. 
11. Notice that original post details screen is displayed.

Test.2
1. Repeat steps 1 to 7 from the previous test.
2. Click cross button on the "Related Post" screen.
3. Notice that original post details screen is displayed skipping any intermediate related post in the back stack (history).

https://user-images.githubusercontent.com/1405144/110086863-eb4bcd00-7db8-11eb-82d5-14345abd7d13.mp4

Review Instructions:
Only 1 reviewer is needed, anyone relatively free can review.

PR submission checklist:
